src/routes/admin/ciraconfig/editCiraConfig.ts

```diff
@@ -37,10 +37,15 @@ export async function editCiraConfig (req, res): Promise<void> {
       // SQL Query > Insert Data
       const results = await ciraConfigDb.updateCiraConfig(ciraConfig)
       if (results !== undefined) {
-        // update the password in Vault if not null
-        if (req.secretsManager && !ciraConfig.generateRandomPassword) {
-          await req.secretsManager.writeSecretWithKey(`${EnvReader.GlobalEnvConfig.VaultConfig.SecretsPath}CIRAConfigs/${ciraConfig.configName}`, ciraConfig.password, mpsPwd)
-          log.info(`MPS password updated in Vault for CIRA Config ${ciraConfig.configName}`)
+        if (req.secretsManager) {
+          if (oldConfig.password != null && ciraConfig.generateRandomPassword) {
+            log.debug('Delete in vault') // User might be flipping from false to true which we dont know. So try deleting either way.
```

Comment:
"Delete in vault" -> "Attempting to delete password from vault"

Reply:
done